packages/contracts/contracts/minter-suite/Minters/MinterSetPriceHolderV5.sol

```diff
@@ -0,0 +1,790 @@
+     * owned by msg.sender being used to prove right to purchase.
+     * @param _ownedNFTTokenId ERC-721 NFT token ID owned by msg.sender being used
+     * to prove right to purchase.
+     * @return tokenId Token ID of minted token
+     */
+    function purchase(
+        uint256 _projectId,
+        address _coreContract,
+        address _ownedNFTAddress,
+        uint256 _ownedNFTTokenId
+    ) external payable returns (uint256 tokenId) {
+        tokenId = purchaseTo(
+            msg.sender,
+            _projectId,
+            _coreContract,
+            _ownedNFTAddress,
+            _ownedNFTTokenId,
+            address(0)
+        );
+        return tokenId;
+    }
+
+    /**
+     * @notice Purchases a token from project `_projectId` and sets
+     * the token's owner to `_to`.
+     * @param _to Address to be the new token's owner.
+     * @param _projectId Project ID to mint a token on.
+     * @param _ownedNFTAddress ERC-721 NFT holding the project token owned by
+     * msg.sender being used to claim right to purchase.
+     * @param _ownedNFTTokenId ERC-721 NFT token ID owned by msg.sender being used
+     * to claim right to purchase.
+     * @return tokenId Token ID of minted token
+     */
+    function purchaseTo(
+        address _to,
+        uint256 _projectId,
+        address _coreContract,
+        address _ownedNFTAddress,
+        uint256 _ownedNFTTokenId
+    ) external payable returns (uint256 tokenId) {
+        return
+            purchaseTo(
+                _to,
+                _projectId,
+                _coreContract,
+                _ownedNFTAddress,
+                _ownedNFTTokenId,
+                address(0)
+            );
+    }
+
+    // public getter functions
+    /**
+     * @notice Gets the maximum invocations project configuration.
+     * @param _contractAddress The address of the contract where the data is stored.
+     * @param _projectId The ID of the project whose data needs to be fetched.
+     * @return MaxInvocationsLib.MaxInvocationsProjectConfig instance with the
+     * configuration data.
+     */
+    function maxInvocationsProjectConfig(
+        address _contractAddress,
+        uint256 _projectId
+    )
+        external
+        view
+        returns (MaxInvocationsLib.MaxInvocationsProjectConfig memory)
+    {
+        return
+            _maxInvocationsProjectConfigMapping[_contractAddress][_projectId];
+    }
+
+    /**
+     * @notice Gets the base project configuration.
+     * @param _contractAddress The address of the contract where the data is stored.
+     * @param _projectId The ID of the project whose data needs to be fetched.
+     * @return ProjectConfig instance with the project configuration data.
+     */
+    function projectConfig(
+        address _contractAddress,
+        uint256 _projectId
+    ) external view returns (ProjectConfig memory) {
+        return _projectConfigMapping[_contractAddress][_projectId];
+    }
```

Comment:
if we add the `struct HolderProjectConfig`, we should add a view for it (analogous to what you did on Merkle minter)

Reply:
@ryley-o i think the view for this data is actually already satisfied by the existing view `allowedProjectHolders`

Reply:
ah, yes! resolving 👍